code/datums/abilities/critter/eat_limb.dm

```diff
@@ -0,0 +1,98 @@
+/datum/targetable/critter/eat_limb
+	name = "Bite Limb"
+	desc = "Swallow a limb on the ground, or attempt to gnaw a random one off of someone!"
+	icon_state = "mimic_eat_limb"
+	cooldown = 45 SECONDS
+	targeted = TRUE
+	target_anything = TRUE
+	cooldown_after_action = TRUE
+
+	cast(atom/target)
+		. = ..()
+		if (istype(target, /obj/item/parts/human_parts))
+			var/obj/item/parts/human_parts/limb = target
+			if (limb.mimic_edible)
+				actions.start(new/datum/action/bar/icon/eat_limb(holder, limb, holder.owner), holder.owner)
+		else if (ishuman(target))
+			var/mob/living/carbon/human/targetHuman = target
+			if (!targetHuman.limbs)
+				boutput(holder.owner, "Your target doesn't have any limbs! Did you do that?")
+				return
+			else
+				var/randLimb = list("random_limb_string" = pick("l_arm", "r_arm", "l_leg", "r_leg"))
+				var/obj/item/parts/human_parts/targetLimb = targetHuman.limbs.get_limb(randLimb["random_limb_string"])
+				if (targetLimb)
+					boutput(world, SPAN_ALERT("<b>[holder.owner] starts to gnaw at [targetLimb]!</b>"))
+					actions.start(new/datum/action/bar/icon/eat_limb(holder, targetLimb, holder.owner, TRUE), holder.owner)
+				else
+					src.cast(target)
```

Comment:
Consider building a list of possible limbs to pick from instead of relying on possibly non-terminating recursion.